### ISSUE TITLE
Move publishStrategy to method argument

### ISSUE THREAD

**Issue by awaniak** [Trendyol/kediatR#88]
Hi
I would want to use the mediator publish method with different publish strategies. Right now it is problematic - one must create different mediator implementations with different publish strategies. I am suggesting adding publish method with publishStrategy argument:
`Mediator`
```kotlin
    suspend fun <T : Notification> publish(
        notification: T,
        publishStrategy: PublishStrategy = StopOnExceptionPublishStrategy()
    )
```

`MediatorImpl`
```kotlin
    override suspend fun <T : Notification> publish(
        notification: T,
        publishStrategy: PublishStrategy,
    ) = processPipeline(
        registry.getPipelineBehaviors(),
        notification
    ) {
        publishStrategy.publish(notification, registry.resolveNotificationHandlers(notification.javaClass))
    }
```

Let me know, what you think
Cheers

**Comment by osoykan** [comment]
Hi @awaniak, that makes sense, indeed. Thanks for the suggestion. Would you like to create a PR for that?

**Comment by awaniak** [comment]
Yeah, sure I will prepare something.

**Comment by awaniak** [comment]
@osoykan Can't push a branch to the repo:
```
remote: Permission to Trendyol/kediatR.git denied to awaniak.
fatal: unable to access 'https://github.com/Trendyol/kediatR.git/': The requested URL returned error: 403
```
I wanted to push my branch and create a PR
Can you help me with this?

**Comment by osoykan** [comment]
@awaniak, the recommended approach is forking the repository and sending a PR afterward. You would have full access to the repository after you fork it.

**Comment by osoykan** [comment]
Implemented with: #92
